src/components/views/location/LocationPicker.tsx

```diff
@@ -20,100 +20,37 @@ import { logger } from "matrix-js-sdk/src/logger";
-import Dropdown from "../elements/Dropdown";
-import LocationShareType from "./LocationShareType";
 import { _t } from '../../../languageHandler';
 import { replaceableComponent } from "../../../utils/replaceableComponent";
 
-interface IDropdownProps {
-    value: LocationShareType;
-    label: string;
-    width?: number;
-    onChange(type: LocationShareType): void;
-}
-
-const LocationShareTypeDropdown = ({
-    value,
-    label,
-    width,
-    onChange,
-}: IDropdownProps) => {
-    const options = [
-        <div key={LocationShareType.Custom}>{
-            _t("Share custom location")
-        }</div>,
-        <div key={LocationShareType.OnceOff}>{
-            _t("Share my current location as a once off")
-        }</div>,
-        // <div key={LocationShareType.OneMin}>{
-        //     _t("Share my current location for one minute")
-        // }</div>,
-        // <div key={LocationShareType.FiveMins}>{
-        //     _t("Share my current location for five minutes")
-        // }</div>,
-        // <div key={LocationShareType.ThirtyMins}>{
-        //     _t("Share my current location for thirty minutes")
-        // }</div>,
-        // <div key={LocationShareType.OneHour}>{
-        //     _t("Share my current location for one hour")
-        // }</div>,
-        // <div key={LocationShareType.ThreeHours}>{
-        //     _t("Share my current location for three hours")
-        // }</div>,
-        // <div key={LocationShareType.SixHours}>{
-        //     _t("Share my current location for six hours")
-        // }</div>,
-        // <div key={LocationShareType.OneDay}>{
-        //     _t("Share my current location for one day")
-        // }</div>,
-        // <div key={LocationShareType.Forever}>{
-        //     _t("Share my current location until I disable it")
-        // }</div>,
-    ];
-
-    return <Dropdown
-        id="mx_LocationShareTypeDropdown"
-        className="mx_LocationShareTypeDropdown"
-        onOptionChange={(key: string) => {
-            onChange(LocationShareType[LocationShareType[parseInt(key)]]);
-        }}
-        menuWidth={width}
-        label={label}
-        value={value.toString()}
-    >
-        { options }
-    </Dropdown>;
-};
-
 interface IProps {
-    onChoose(
-        uri: string,
-        ts: number,
-        type: LocationShareType,
-    ): boolean;
+    onChoose(uri: string, ts: number): boolean;
     onFinished(ev?: SyntheticEvent): void;
 }
 
 interface IState {
-    type: LocationShareType;
     position?: GeolocationPosition;
-    manualPosition?: GeolocationPosition;
     error: Error;
 }
 
+/*
+ * An older version of this file allowed manually picking a location on
+ * the map to share, instead of sharing your current location.
+ * Since the current designs do not cover this case, it was removed from
+ * the code but you should be able to find it in the git history by
+ * searching for the commit that remove manualPosition from this file.
+ */
```

Comment:
is this intending to document away any remaining oddities that were left behind (potentially) unintentionally? Not really sure what the comment is trying to inform the reader of.

Reply:
It's intended to help someone in the future resurrect the code, when we do want that functionality.